pkg/operator/options/options.go

```diff
@@ -44,6 +53,23 @@ type Options struct {
 	EnableLeaderElection bool
 	MemoryLimit          int64
 	LogLevel             string
+	BatchMaxDuration     time.Duration
+	BatchIdleDuration    time.Duration
+	FeatureGates         FeatureGates
+}
+
+// Note: temporary flags to note if merged fields have been set
+type optionsFlags struct {
+	BatchMaxDurationSet  bool
+	BatchIdleDurationSet bool
+	FeatureGatesSet      bool
+}
+
+type Options struct {
+	OptionFields
+	optionsFlags
+
+	featureGateStr string
```

Comment:
Does it make sense to store featureGatesStr as an unexported element under the featureGates in the struct?